org.eclipse.richbeans.test/src/org/eclipse/richbeans/test/shuffle/TestUI.java

```diff
@@ -0,0 +1,82 @@
+package org.eclipse.richbeans.test.shuffle;
+
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+
+/**
+ * 
+ * Class used to start and stop an SWT Display for use with the test.
+ * 
+ * @author Matthew Gerring
+ *
+ */
+class TestUI {
+
+	private ReentrantLock     currentTestLock;
+	private final CyclicBarrier swtBarrier = new CyclicBarrier(2);
+	private Thread uiThread;
+	
+	private ShellTest currentTest;
+	private volatile boolean disposed = false;
+	private Shell appShell;
+	
+	public void start()  {
+		
+		currentTestLock = new ReentrantLock();
+		currentTestLock.lock();
+		uiThread = new Thread(new Runnable() {
+
+			public void run() {
+				try {
+					System.out.println("Starting "+Thread.currentThread().getName());
+					while (!disposed && !uiThread.isInterrupted()) {
+						
+						currentTestLock.lockInterruptibly();

```

Comment:
I'm not clear on what this is doing - you lock then unlock immediately afterwards? Or is this to cause allow a wait on a  call from the createBot/disposeBot methods?